src/interface/storage.thrift

```diff
@@ -557,24 +557,29 @@ struct LookupAndTraverseRequest {
  * End of Index section
  */
 
+struct ScanCursor {
+    3: bool                                 has_next,
+    // next start key of scan, only valid when has_next is true
+    4: optional binary                      next_cursor,
+}
+
 struct ScanVertexRequest {
```

Comment:
@jievince @Nicole00 PTAL, do we need to keep the old interface? It depends on how we use them. Do we scan in  parallel(code level) or scan for each part?

Reply:
I suggest to keep the old interface, StorageClient provides interface to scan just one part. And Connectors depends on this kind of interface.

Reply:
Youd can't just scan one partition, just wrap the origin partition ID to list

Reply:
> Youd could just scan one partition, just wrap the origin partition ID to list

The hard part is the cursor, some part may have scanned all data, while others still has more. I guess this need some extra code to do it.

How about we only change the code from
```
2: common.PartitionID part_id,
3: optional binary cursor,
```
to
```
2: list<common.PartitionID> part_id,
3: optional list<binary> cursor,
```

In that way, the java/python client could still only support the old interface `scan(partId)` to users, but only convert to a `list<partId>` when sending RPC. I think this change is the minimum one?

Reply:
> @jievince @Nicole00 PTAL, do we need to keep the old interface? It depends on how you use them. Do we scan in parallel(code level) or scan for each part one by one?

I use the old scan interface to scan in parallel: I start multiple threads, each thread to scan a part

Reply:
@critical27 When partition scan to end, its cursor will set to null.

Reply:
@Shylock-Hg @critical27  There's one `incompatibility` problem if we don't keep the origin interfaces.  Change the partition id from int32 to list when sending RPC request will cause the incompatibility, and Nebula Server will not support **Backward Compatible**.

Reply:
> @Shylock-Hg @critical27 There's one `incompatibility` problem if we don't keep the origin interfaces. Change the partition id from int32 to list when sending RPC request will cause the incompatibility, and Nebula Server will not support **Backward Compatible**.

I mean the client could still use the origin `user interface` (not the rpc interface, I mean sth like `scan(spaceId, partId)` in java client), but wrap into the new rpc interface using old user interface. It is ok for incompatibility in rpc layer for 1230 version. But the user interface better keep the same.

Reply:
Yes, we had agree to this after offline discussion

Reply:
👍